ranch/transit.py

```diff
@@ -594,17 +626,26 @@ def snap_stop_to_node(self):
         RanchLogger.info('Snapping gtfs stops to roadway node...')
 
         # get rid of motorway nodes
-        non_motorway_links_df = self.roadway_network.links_df[
-            ~self.roadway_network.links_df.roadway.isin(["motorway", "motorway_link"])
+        motorway_links_df = self.roadway_network.links_df[
+            self.roadway_network.links_df.roadway.isin(["motorway", "motorway_link", 'trunk', 'trunk_link'])
         ].copy()
+        motorway_nodes_list = motorway_links_df.fromIntersectionId.tolist() + motorway_links_df.toIntersectionId.tolist()
+
+        ag100plus_links_df = self.roadway_network.links_df[
+            self.roadway_network.links_df.assign_group>=100
+        ].copy()
+        ag100plus_nodes_list = ag100plus_links_df.fromIntersectionId.tolist() + ag100plus_links_df.toIntersectionId.tolist()
+
+        bus_only_links_df = self.roadway_network.links_df[
+            self.roadway_network.links_df.bus_only==1
+        ].copy()
+        bus_only_nodes_list = bus_only_links_df.fromIntersectionId.tolist() + bus_only_links_df.toIntersectionId.tolist()
 
         node_candidates_for_stops_df = self.roadway_network.nodes_df[
             (
-                self.roadway_network.nodes_df.shst_node_id.isin(
-                    non_motorway_links_df.fromIntersectionId.tolist()
-                    + non_motorway_links_df.toIntersectionId.tolist()
-                )
-                & (self.roadway_network.nodes_df.drive_access == 1)
+                ((~self.roadway_network.nodes_df.shst_node_id.isin(motorway_nodes_list + ag100plus_nodes_list)) &
+                (self.roadway_network.nodes_df.drive_access == 1)) | 
+                (self.roadway_network.nodes_df.shst_node_id.isin(bus_only_nodes_list))
```

Comment:
Suggest reverting back to the old algorithm.
1. Get subset of network that excludes LINKS that are a) motorways, b) trunks, c) their ramps, d) ag100+, and e) drive_access=0 that are not bus_only.
2.  Then use nodes from this subset of links as candidates for stops.